packages/orm/src/EntityManager.ts

```diff
@@ -1260,6 +1262,8 @@ export class EntityManager<C = unknown, Entity extends EntityW = EntityW> {
           if (e.isNewEntity && !e.isDeletedEntity) this.#entityIndex.set(e.idTagged, e);
           getInstanceData(e).resetAfterFlushed();
         }
+        // Update the joinRows refs to reflect the new state
+        allFlushedJoinRowsResetFns.forEach((resetFn) => resetFn());
```

Comment:
I'd prefer something like:

```
for (joinRow in joinRowTodos) {
  joinRow.resetAfterFlushed();
}
```

Where `JoinRow.resetAfterFlushed` changed did something like `this.op: "inserted"` or `this.op = "deleted"`, similar to how `InstanceData.resetAfterFlushed` works.

And then ideally `JoinRow.hasChanges` changes from:

```
  get hasChanges() {
    const todos = this.toTodo();
    return !!todos;
  }
```

To more like `this.op = pending || flushed`

I.e. `JoinRow.op` is a literal of pending/inserted/deleted (or whatever ~2-3 values make sense to handle both "pending insert" + "pending delete" + "flushed").